### ISSUE TITLE
Can't run OCBuilder - stops at Compile

### ISSUE THREAD

**Issue by L0rdLogan** [Pavo-IM/ocbuilder#28]
Old one got closed down, but finally got around to running it in Terminal like asked

Andrews-iMac:~ andrew$ /Users/andrewfishburn/Downloads/ocbuilder-master/OCBuilder/release.command ; exit;
mkdir: /OCBuilder_Clone: Read-only file system
/Users/andrew/Downloads/ocbuilder-master/OCBuilder/release.command: line 275: cd: /OCBuilder_Clone: No such file or directory
Cloning Lilu repo...
Cloning into 'Lilu'...
remote: Enumerating objects: 3, done.
remote: Counting objects: 100% (3/3), done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 4029 (delta 0), reused 0 (delta 0), pack-reused 4026
Receiving objects: 100% (4029/4029), 4.23 MiB | 883.00 KiB/s, done.
Resolving deltas: 100% (2452/2452), done.
/Users/andrew/Downloads/ocbuilder-master/OCBuilder/release.command: line 279: cd: /OCBuilder_Clone/Lilu: No such file or directory
Compiling the latest commited Debug version of Lilu...
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance
logout
Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

[Process completed]

GUI version just stops when trying to Compile, even with 7.4

I have no idea what I am doing when it comes to Xcode, hence using this program to do it for me

Also using the command line version makes a mess out of the User folder, please adjust it so it all goes into a folder in your user directory

States it goes into /OCbuilder_Clone, but I can't locate this folder

GUI:
![Greenshot 2020-07-15 20 02 15](https://user-images.githubusercontent.com/5092548/87584921-27210d80-c6d6-11ea-8f4f-97d5f5100df3.png)

**Comment by Squirrelf** [comment]
Same issue for me, I'm on Big Sure Beta 2

TLDR version of how I worked around it:
install Xcode if you don't have it
sudo xcode-select -s /Applications/Xcode-beta.app/Contents/Developer (this tells it to use the full Xcode instead of just the command line tools)
download OpenCorePkg source code from GitHub run build_oc.tool. (this will install MTOC and I believe NASM, if NASM is not installed then install it with homebrew: brew install nasm)
and OCBuilder worked for me (I only tried release, but I assume debug will work just as well)

---- long version as to how I got there:

same issue with v11.5 and v12 beta2 of the command line tools

I also tried xcode-select on the full Xcode
sudo xcode-select -s /Applications/Xcode.app/Contents/Developer
sudo xcode-select -s /Applications/Xcode-beta.app/Contents/Developer

and it did go further, but ending up with "NASM missing, installing...", it didn't seem like it was doing any installing, so I installed it using home-brew (brew install nasm)
then re-ran OCBuilder, this time said "MTOC missing, installing...", but doesn't gets installed.
Then remember that OpenCorePkg source code builder will install those tools, 
so I downloaded OpenCorePkg source code and ran build_oc.tool (found in the source code of OpenCorePkg) which will install MTOC and possibly NASM, but already installed it with homebrew

Finally, re-ran OCbuild tool and it worked.

**Comment by iGPU** [comment]
I see same thing as L0rdLogan: unable to compile kext files. 

However, if 'With Kexts?" is un-checked, OC compiles just fine. So seems to be a problem with kext compilation.

**Comment by Pavo-IM** [comment]
First of all I would remind you to read the README, the app requires the "FULL" version of Xcode to be installed and the EULA to be accepted. This will install the commandline tools directly from that version of Xcode. You can not just install commandline tools only. The app also checks for NASM and MTOC if its is currently installed and if not, prompts for administrative rights to install those for you. You can not use home-brew version of NASM or MTOC, OpenCorePkg build tool itself checks for a specific version and if it doesn't match it will prompt to install that version, which is the same version I use in my app. The app uses the /tmp/OCBuilder_Clone/ directory for its cloning and compiling directory, it does not use User directory. I have ran the app in Release w/Kext, Release without/Kext and Debug w/Kext and Debug without/Kext and everything is cloning and compiling as it should. It has to do with your system configuration on why it's not working correctly. I would suggest reading the README and installing the "FULL" version of Xcode without using home-brew command line tools or NASM or MTOC versions.
As you can see below it is going past Lilu:
![Screenshot 2020-07-16 12 12 53](https://user-images.githubusercontent.com/33814640/87696298-9b9f8f00-c75e-11ea-862b-3dfc7acd1859.png)
![Screenshot 2020-07-16 12 18 54](https://user-images.githubusercontent.com/33814640/87696323-a2c69d00-c75e-11ea-9383-cd52d4e2ae9c.png)

**Comment by Pavo-IM** [comment]
> ![Greenshot 2020-07-15 20 02 15](https://user-images.githubusercontent.com/5092548/87584921-27210d80-c6d6-11ea-8f4f-97d5f5100df3.png)

Try selecting your Desktop or another location for your destination build folder, seems like the directory you have selected, your user account doesn't have full access too.

**Comment by L0rdLogan** [comment]
I am running the full version of Xcode and that OCBuilder folder is on my desktop and has full write access.

I have re-installed XCode - and no difference unfortunately

![image](https://user-images.githubusercontent.com/5092548/87697950-54c39080-c782-11ea-9abe-0b68519ce796.png)

**Comment by Pavo-IM** [comment]
> I am running the full version of Xcode and that OCBuilder folder is on my desktop and has full write access.
> 
> I have re-installed XCode - and no difference unfortunately
> 
> ![image](https://user-images.githubusercontent.com/5092548/87697950-54c39080-c782-11ea-9abe-0b68519ce796.png)

Open Xcode, go to Xcode > Preferences > Location tab and show me a picture of that setting like this:
![Screenshot 2020-07-16 12 37 10](https://user-images.githubusercontent.com/33814640/87698236-35683b80-c761-11ea-8cd7-274088ace403.png)

**Comment by L0rdLogan** [comment]
Ah, that solved the issue! I had no command line tools selected as per your picture. I woulsn't even have thought to have looked there.

thank you

**Comment by Pavo-IM** [comment]
Commandline Tools in that section or automatically set when you accept the EULA of Xcode, and if its glitches out sometimes you can always do it by commandline `sudo xcode-select -s /Applications/Xcode.app/Contents/Developer`. No problem though, glad you got it sorted out, closing the issue.
